product-service/src/libs/api-gateway.ts

```diff
@@ -0,0 +1,25 @@
+import type { APIGatewayProxyEvent, APIGatewayProxyResult, Handler } from "aws-lambda"
+import type { FromSchema } from "json-schema-to-ts";
+
+export type ValidatedAPIGatewayProxyEvent<S> = Omit<APIGatewayProxyEvent, 'body'> & { body: FromSchema<S> }
+export type ValidatedEventAPIGatewayProxyEvent<S> = Handler<ValidatedAPIGatewayProxyEvent<S>, APIGatewayProxyResult>
+
+export const formatJSONResponse = (response: Record<string, unknown>) => {
+  return {
+    statusCode: 200,
+    headers: {
+      'Access-Control-Allow-Origin': '*',
+    },
+    body: JSON.stringify(response)
+  }
+};
+
+export const formatErrorJSONResponse = (response: Record<string, unknown>) => {
+  return {
+    statusCode: 500,
```

Comment:
From my point of view 500 (Internal Server Error) is incorrect response status code for these cases. I suggest to use 404 in case of 'No product with id...' and 422 in another one. This doesn't affect the score, I think in this task we check other aspects, but I think that suitable response status codes is important part of REST API.